### ISSUE TITLE
input buffer sometimes incorrectly marked as inplace when it's not in Inductor

### ISSUE THREAD

**Issue by Chillee** [pytorch/pytorch#120217]
### 🚀 The feature, motivation and pitch

```
import torch
import torch.nn as nn

D = 4096
B = 2048*32
inp = torch.randn(B, D, dtype=torch.float16)
gO = torch.randn(B, D, dtype=torch.float16)
mod = nn.LayerNorm([D], dtype=torch.float16)

comp_mod = torch.compile(mod, dynamic=False)
print(D)
comp_mod(inp)
```

Generates https://pastebin.com/cJZbb5VY

In particular, ` tl.store(in_out_ptr0 + (x0), tmp10, None)` despite the fact that `in_out_ptr0` is never read from.

I suspect that in certain cases, when we fuse scheduler nodes reads that should be dropped aren't dropped?

cc: @peterbell10 @eellison @shunti

cc @ezyang @bdhirsh @anijain2305 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @msaroufim @zou3519 @aakhundov

**Comment by eellison** [comment]
It shouldn't be an in_out_ptr (just an output) but `buf3` is a returned output so it shouldn't be dropped.

**Comment by eellison** [comment]
Are you working on this? if not, we can re-assign?

**Comment by masnesral** [comment]
@eellison , I am not. If it's bad to have it languishing in my queue, fine with me if someone else has bandwidth.
